src/components/ComposerQuill/composer.js

```diff
@@ -155,6 +201,94 @@ class Composer extends React.Component {
     this.quill.insertText(range.index, text);
   }
 
+  // converts a string of text into operation deltas
+  buildContents(text) {
+    const split = text.split(/(<spark-mention [a-zA-Z0-9-='"\s]+>.+?<\/spark-mention>)/);
+
+    const contents = split.map((line) => {
+      // convert spark-mention into a mention delta
+      if (line.indexOf('<spark-mention ') === 0) {
+        // converts the string to a html element so we can grab the data
+        const object = new DOMParser().parseFromString(line, 'text/xml');
+        // DOMParser returns a document but we just need the first element
+        const mention = object.firstChild;
+        const objectType = mention.getAttribute('data-object-type');
+        let objectId;
+
+        if (objectType === 'person') {
+          objectId = mention.getAttribute('data-object-id');
+        }
+
+        return {
+          insert: {
+            mention: {
+              index: 0,
+              denotationChar: '@',
+              id: objectId,
+              objectType,
+              value: mention.textContent,
+            },
+          },
+        };
+      }
+
+      // otherwise just insert the text
+      return {insert: line};
+    });
+
+    return contents;
+  }
+
+  renderMentionItem(item) {
+    const {id, src, displayName} = item;
+    let classes = 'ql-mention-avatar';
+    let avatar;
+    let secondary;
+
+    if (src) {
+      // if we have a picture then use that
+      avatar = `<img class='${classes}' src='${src}'>`;
+    } else {
+      // otherwise we build it ourself
+      let initials;
+
+      if (id === 'all') {
+        // avatar is a circle @ for all
+        classes += ' all';
+        initials = '@';
+        secondary = 'Mention everyone in this space';
+      } else {
+        // use the initials of the name as the avatar
+        let chars = displayName.charAt(0);
+        const space = displayName.indexOf(' ');
+
+        if (space >= 0) {
+          chars += displayName.charAt(space + 1);
+        }
+
+        initials = chars.toUpperCase();
+      }
+
+      avatar = `<div class='${classes}'>${initials}</div>`;
```

Comment:
This could probably go in a util file to keep this one a little smaller. Maybe group some of the other utils too.